File: java/src/juegoTexto/Controlador.java
package juegoTexto;

import civitas.CivitasJuego;
import civitas.GestionesInmobiliarias;
import civitas.OperacionesJuego;
import civitas.Respuestas;
import civitas.SalidasCarcel;
import civitas.OperacionInmobiliaria;

class Controlador {
    private CivitasJuego juego;
    private VistaTextual vista;

    Controlador(CivitasJuego juego, VistaTextual vista) {
        this.juego = juego;
        this.vista = vista;
    }

    void juega() {
        vista.setCivitasJuego(juego);
        while (!juego.finalDelJuego()) {
            vista.pausa();
            OperacionesJuego siguiente=juego.siguientePaso();
            vista.mostrarSiguienteOperacion(siguiente);
            if (siguiente!=OperacionesJuego.PASAR_TURNO) vista.mostrarEventos();

            if (juego.finalDelJuego())
                juego.actualizarInfo();
            else {
                switch (siguiente) {
                    case COMPRAR:
                        Respuestas rp=vista.comprar();
                        if(rp==Respuestas.SI){
                            juego.comprar();
                        }
                        juego.siguientePasoCompleado(OperacionesJuego.COMPRAR);
                        break;

                    case GESTIONAR:
                        vista.gestionar();
                        OperacionInmobiliaria operacion=new OperacionInmobiliaria(GestionesInmobiliarias.values()[vista.getGestion()], vista.getPropiedad());
                        int ip=operacion.getNumPropiedad();
                        switch(operacion.getGestion()){
                            case VENDER:
                                juego.vender(ip);
                                break;
                            case HIPOTECAR:
                                juego.hipotecar(ip);
                                break;
                            case CANCELAR_HIPOTECA:
                                juego.cancelarHipoteca(ip);
                                break;
                            case CONSTRUIR_CASA:
                                juego.construirCasa(ip);
                                break;
                            case CONSTRUIR_HOTEL:
                                juego.construirHotel(ip);
                                break;
                            default:
                                break;
                        }
                        juego.siguientePasoCompleado(OperacionesJuego.GESTIONAR);
                        break;

                    case SALIR_CARCEL:
                        SalidasCarcel salida=vista.salirCarcel();
                        switch (salida) {
                            case PAGANDO:
                                juego.salirCarcelPagando();
                                juego.siguientePasoCompleado(OperacionesJuego.SALIR_CARCEL);
                                break;
                        
                            case TIRANDO:
                                juego.salirCarcelTirando();
                                juego.siguientePasoCompleado(OperacionesJuego.SALIR_CARCEL);
                                break;
                        }

                        break;
                
                    default:
                        break;
                }

                vista.actualizarVista();
            }
        }
    }
}

File: ruby/lib/controlador.rb
require_relative('operacion_inmobiliaria')
require_relative('gestiones_inmobiliarias')

module Civitas
    class Controlador
        def initialize(juego, vista)
            @juego = juego #CivitasJuego
            @vista = vista #VistaTextual
        end

        def juega #void
            
            @vista.setCivitasJuego(@juego)

            isFinal = @juego.finalDelJuego()
            while !isFinal do
                @vista.pausa
                siguientePaso = @juego.siguientePaso
                @vista.mostrarSiguienteOperacion(siguientePaso)
                
                
                if (siguientePaso != Operaciones_juego::PASAR_TURNO)
                    @vista.mostrarEventos
                end
                
                isFinal = @juego.finalDelJuego()

                if (!isFinal)
                    puts siguientePaso.inspect

                    case siguientePaso
                    when Operaciones_juego::COMPRAR
                        if (@vista.comprar == Respuestas::SI)
                            @juego.comprar
                        end
                        @juego.siguientePasoCompletado(Operaciones_juego::COMPRAR)
                                                            
                    when Operaciones_juego::GESTIONAR
                        @vista.gestionar
                        operacion = OperacionInmobiliaria.new(@vista.getGestion, @vista.getPropiedad)
                        ip=operacion.numPropiedad

                        case operacion.gestion
                        when GestionesInmobiliarias::VENDER
                            @juego.vender(ip)
                        when GestionesInmobiliarias::HIPOTECAR
                            @juego.hipotecar(ip)
                        when GestionesInmobiliarias::CANCELAR_HIPOTECA
                            @juego.cancelarHipoteca(ip)
                        when GestionesInmobiliarias::CONSTRUIR_CASA
                            @juego.construirCasa(ip)
                        when GestionesInmobiliarias::CONSTRUIR_HOTEL
                            @juego.construirHotel(ip)
                        end
                        @juego.siguientePasoCompletado(Operaciones_juego::GESTIONAR)
                                       
                    when Operaciones_juego::SALIR_CARCEL
                        
<<<<<<< HEAD
                    else
                        puts "No esta entrando por ningún lado" 
=======
                        case @vista.salirCarcel
                        when SalidasCarcel::PAGANDO
                            @juego.salirCarcelPagando
                            @juego.siguientePasoCompletado(Operaciones_juego::SALIR_CARCEL)
                        when SalidasCarcel::TIRANDO
                            @juego.salirCarcelTirando
                            @juego.siguientePasoCompletado(Operaciones_juego::SALIR_CARCEL)
                        end
>>>>>>> 5011d6f146ad65ecc16fa8dfe38d54b2def790d4
                    end
                    @vista.actualizarVista
                else
                    @juego.actualizarInfo
                end
            end       
        end
    end
end